crates/configuration/src/shared/types.rs

```diff
@@ -38,6 +51,32 @@ impl Serialize for U128 {
     }
 }
 
+impl<'de> Deserialize<'de> for U128 {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct U128Visitor;
+
+        impl<'de> de::Visitor<'de> for U128Visitor {
+            type Value = U128;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("an integer between 0 and 2^128 − 1.")
+            }
+
+            fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                Ok(U128(v.to_string().parse::<u128>().unwrap()))
```

Comment:
With above modification we get:

```suggestion
                v.try_into().map_err(de::Error::custom)
```